### PR TITLE
[REVIEW] Upgrade to libcu++ on GitHub

### PR DESCRIPTION
**Pull request by trxcllnt**
Update to libcu++ on Github.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/include/cuspatial/cubic_spline.hpp [hunk 1]
```diff
@@ -18,7 +18,10 @@
 
 #include <cudf/column/column.hpp>
 #include <cudf/table/table.hpp>
+#include <cudf/types.hpp>
+
 #include <memory>
+#include "cudf/types.hpp"
```
Comment (harrism) [inline cpp/include/cuspatial/cubic_spline.hpp:24]: clangd is messing with you.

```suggestion
```

File: cpp/src/indexing/construction/detail/utilities.cuh [hunk 2]
```diff
@@ -19,8 +19,10 @@
 #include <cudf/column/column_factories.hpp>
 #include <cudf/types.hpp>
 #include <cudf/utilities/type_dispatcher.hpp>
+#include "rmm/cuda_stream_view.hpp"
```
Comment (harrism) [inline cpp/src/indexing/construction/detail/utilities.cuh:22]:
```suggestion
```

File: cpp/src/join/detail/traversal.cuh [hunk 3]
```diff
@@ -20,6 +20,7 @@
 #include "utility/z_order.cuh"
```
Comment (harrism) [inline cpp/src/join/detail/traversal.cuh:20]:
```suggestion
#include <utility/z_order.cuh>
```

File: cpp/src/spatial/hausdorff.cu [hunk 4]
```diff
@@ -14,23 +14,25 @@
  * limitations under the License.
  */
 
+#include "utility/scatter_output_iterator.cuh"
+#include "utility/size_from_offsets.cuh"
```
Comment (harrism) [inline cpp/src/spatial/hausdorff.cu:18]:
```suggestion
#include <utility/scatter_output_iterator.cuh>
#include <utility/size_from_offsets.cuh>
```

**Review by harrism — CHANGES_REQUESTED** [review]
Just some header stuff.

**Comment by trxcllnt** [comment]
@harrism looks like the base was set to branch-0.16 for some reason. Changing it invalidated some of your suggestions, but I'll merge the rest.

**Comment by trxcllnt** [comment]
rerun tests

**Comment by trxcllnt** [comment]
rerun tests

**Review by harrism — APPROVED** [review]

**Review by cwharris — APPROVED** [review]

**Pull request merged**